backend/src/connectors/store.py

```
@@ -30,6 +31,20 @@
+def get_config_hash() -> str:
+    """
+    Generate a stable hash of the chunking configuration.
+
+    If chunk_size or chunk_overlap changes, the hash changes and a full rebuild is triggered.
+    """
+    config = (
+        f"chunk_size={DOCUMENT_SPLITTER._chunk_size}"
+        f"_overlap={DOCUMENT_SPLITTER._chunk_overlap}"
+        "_tokenizer=unicode_v1"
+    )
+    return hashlib.md5(config.encode()).hexdigest()[:8]
```

Ruff (0.14.14)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)